firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallationException.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations;
+
+import androidx.annotation.NonNull;
+import com.google.firebase.FirebaseException;
+
+/** The class for all Exceptions thrown by {@link FirebaseInstallations}. */
+public class FirebaseInstallationException extends FirebaseException {
+
+  public enum Status {
+    SDK_INTERNAL_ERROR,
+
```

Comment:
I suggest to sync with mertcan@ or Rayo about what status we want to expose to dev

Reply:
@andirayo or @mmermerkaya  Whats the high level exception code to be shown to dev in case FIS requests fail? Currently, I have  SDK_INTERNAL_ERROR.